Userland/Libraries/LibWeb/DOM/Element.h

```diff
@@ -91,7 +91,7 @@ class Element
     // NOTE: This is for the JS bindings
     DeprecatedFlyString namespace_uri() const { return namespace_(); }
 
-    bool has_attribute(DeprecatedFlyString const& name) const;
+    bool has_attribute(StringView name) const;
```

Comment:
Very often when we do this, we end up sliding backwards on performance.
Comparing two `DeprecatedFlyString` against each other is O(1) while comparing something against a `StringView` is O(n).
Ultimately, `has_attribute()` should be taking a `FlyString const&`, but I'm concerned that we're leaving a bunch of these performance regression all over the place since there's no FIXME or anything here about how this should eventually take `FlyString`.

Reply:
Oops, I didn't actually mean to push that set of changes in this MR - only meant to have this one for the very first commit. Was meaning to push all that into another branch in a follow up.

Definitely agree that ultimately this should take a `FlyString const&`. I don't think in this case we're introducing a performance regression since it defers to `NamedNodeMap::get_attribute`, which already takes a `StringView`, so we're dropping that `O(1)` comparison on the floor immediately.

Maybe the correct place to leave such a `FIXME` for now would be in `NamedNodeMap`?

Reply:
Fair enough about `NamedNodeMap` (which should indeed also take `FlyString const&` since we'll be asked about known common attribute names in the vast majority of cases)

I'd say slap FIXMEs on all the APIs :)

Reply:
Will do :^)

Reply:
Done :^)